### PR TITLE
EZP-27455: Doc files proofed

### PR DESCRIPTION
**Pull request by DominikaK**
Language and cosmetic changes to doc files.

### CODE REVIEW

**Comment by DominikaK** [comment]
ping @bdunogier @mere-teresa

File: doc/assets.md [hunk 1]
```diff
@@ -10,41 +10,37 @@ For assets, a special **`ezdesign` asset package** is available.
...
+- `<bundle_directory>/Resources/public/themes/bar/` => Assets will be part of the `bar` theme.
+- `web/assets/themes/biz/` => Assets will be part of the `biz` theme.
 
-It is also possible to use `web/assets` as a global override directory. 
-If called asset is present **directly under this directory**, it will always be considered first.
+It is also possible to use `web/assets` as a global override directory. If called asset is present **directly under this directory**, it will always be considered first.
 
-> **Important**: You must have *installed* your assets with `assets:install` command, so that your public resources are
-> *installed* into the `web/` directory.
+> **Important**: You must have *installed* your assets with `assets:install` command, so that your public resources are *installed* into the `web/` directory.
```
Comment (bdunogier) [inline doc/assets.md]: Those lines are really long. Given that this change doesn't change anything to the rendered result, I suggest you leave it as it was.

File: doc/assets.md [hunk 2]
```diff
@@ -10,41 +10,37 @@ For assets, a special **`ezdesign` asset package** is available.
...
+
+The default fallback order is:
+- Application assets directory: `web/assets/`
+- Application theme directory: `web/assets/themes/<theme_name>/`
+- Bundle theme directory: `web/bundles/<bundle_directory>/themes/<theme_name>/`
 
 Calling `asset("js/foo.js", "ezdesign")` can for example be resolved to `web/bundles/app/themes/my_theme/js/foo.js`.
 
-## Performance & assets resolution
-When using themes, paths for assets are resolved at runtime. This is due to how Symfony Asset component is integrated
```
Comment (bdunogier) [inline doc/assets.md]: Well, I'm sorry but I'm against those changes.
We follow recommendations in regards to lines length (120 usually). It is easier to work with shorter lines in editors, and easier to compare when diffing (like here).

Given that a line break in markdown requires afaik a blank like, those changes don't really improve anything.

Comment (lolautruche) [inline doc/assets.md]: I second this

Comment (kusztel) [inline doc/assets.md]: I would like to add a few things to this. Firstly, which recommendations to follow you mentioned? I do not know about any document with recommendations for writing MD doc files at eZ. Afaik, one of the topics on doc migration is to create set of rules and recommendations for future doc creators/editors. Please, also take a note that doc content is fairly different than code, comments or readme/install files. 
While I'm at readme/install files topic - please take a look at https://github.com/ezsystems/ezplatform-ee-demo/blob/master/INSTALL.md there are so many breaklines inside bullet lists, links and blocks that sometimes few paragraphs can break because of one small change. 
Making changes in such "chopped" content is also a pain to do review later. Instead of 3-4 lines to check, reviewer would have to check dozens of lines and make sure that everything is clean and tidy, because we don't want strangely looking content after content is generated. 
Breaking the lines at 100-120 may seem nice and tidy from developer point of view, but this is documentation, and we prefer to concern about our reader to get proper content fast  - we can scroll a few inches to the right if that's what it takes.

Comment (DominikaK) [inline doc/assets.md]: One more thing to @kusztel's comment, here is the issue for doc migration he mentioned: https://jira.ez.no/browse/EZP-27431 where we propose an internal markdown style guide we want to use for future docs. I applied this style guide here.
While I agree that breaking lines semantically (after a few sentences) is good practice in significantly longer paragraphs, sticking to a fixed width is going to cause problems when future changes to doc files inevitably occur.

Comment (bdunogier) [inline doc/assets.md]: > Breaking the lines at 100-120 may seem nice and tidy from developer point of view, but this is documentation, and we prefer to concern about our reader to get proper content fast - we can scroll a few inches to the right if that's what it takes.

No. The fact that many break lines inside bullets breaks something doesn't mean that we won't use breaklines.

The point of markdown is to be humanly readable. Auto-wrapping of lines is something developers never do.

The point of having this_developer documentation_ as markdown is that _developers_ can contribute to it.

> Making changes in such "chopped" content is also a pain to do review later. Instead of 3-4 lines to check, reviewer would have to check dozens of lines and make sure that everything is clean and tidy

Wrong. It makes it harder. The human brain is built to process short chunks of text, not long lines. Each change in the same line will move the comparison left or right, making it harder to review the longer the line is. Most review tools do support side-by-side reviews, in which case long lines require permanent left-right scrolling.

A line break does not create a line break in the rendered result. A new line does. It's a _feature_ of markdown.

File: doc/assets.md [hunk 3]
```diff
@@ -10,41 +10,37 @@ For assets, a special **`ezdesign` asset package** is available.
 <img src="{{ asset("images/foo.png", "ezdesign") }}" alt="foo"/>
 ```
 
-Using `ezdesign` package will resolve current design with theme fallback.
+Using the `ezdesign` package will resolve current design with theme fallback.
 
 By convention, an asset theme directory can be located in:
-* `<bundle_directory>/Resources/public/themes/`
-* `web/assets/themes/`
+- `<bundle_directory>/Resources/public/themes/`
```
Comment (bdunogier) [inline doc/assets.md]: Why did you change the character used for bullet points ? Both produce exactly the same result.

Comment (DominikaK) [inline doc/assets.md]: With the perspective of moving all doc to GitHub, we are trying to formulate md standards to use for doc which, while not binding for contributors, would make maintaining and editing the doc easier. This was also the reason for making the changes to line lengths.
However, if this is not accepted, I'll roll back the changes.

Comment (lolautruche) [inline doc/assets.md]: While it does produce the same result, I'm fine with this change

Comment (bdunogier) [inline doc/assets.md]: > would make maintaining and editing the doc easier.

Consistency across everything doesn't really make editing and maintaining easier here. What matters is that the same one is used over documents, I think. I'm more okay with that one than with the line length, though.

Just remember that any change is logged, and doing that makes you the last author of the change, making it more complicated to trace down which change a line came from. If after this change somebody does a blame on the file, your change will displayed, while it is not when the content was introduced.

If consistency on bullet points is important, then it should be written down somewhere, and it should be easy to find for doc contributors / developers.

File: doc/assets.md (hunk 2 above)
Comment (bdunogier) [inline doc/assets.md]: > sticking to a fixed width is going to cause problems when future changes to doc files inevitably occur.

It's not a "fixed" length, it's a maximum to be respected within reason.
I still do not see why using unique lines makes things easier, and I have given arguments about why.

File: doc/assets.md [hunk 4]
```diff
@@ -10,41 +10,43 @@ For assets, a special **`ezdesign` asset package** is available.
...
+- `<bundle_directory>/Resources/public/themes/foo/` => Assets will be part of the `foo` theme.
+- `<bundle_directory>/Resources/public/themes/bar/` => Assets will be part of the `bar` theme.
+- `web/assets/themes/biz/` => Assets will be part of the `biz` theme.
 
-It is also possible to use `web/assets` as a global override directory. 
+It is also possible to use `web/assets` as a global override directory.
 If called asset is present **directly under this directory**, it will always be considered first.
 
 > **Important**: You must have *installed* your assets with `assets:install` command, so that your public resources are
-> *installed* into the `web/` directory.
```
Comment (bdunogier) [inline doc/assets.md]: I'd also keep the wrapping of lines here. The rendered result is identical.

File: doc/assets.md [hunk 5]
```diff
@@ -10,41 +10,43 @@ For assets, a special **`ezdesign` asset package** is available.
...
+- Bundle theme directory: `web/bundles/<bundle_directory>/themes/<theme_name>/`
 
 Calling `asset("js/foo.js", "ezdesign")` can for example be resolved to `web/bundles/app/themes/my_theme/js/foo.js`.
 
-## Performance & assets resolution
-When using themes, paths for assets are resolved at runtime. This is due to how Symfony Asset component is integrated
-with Twig. This can cause significant performance impact because of I/O calls when looping over all potential themes
+## Performance and asset resolution
+
+When using themes, paths for assets are resolved at runtime. This is due to how the Symfony Asset component is integrated
```
Comment (bdunogier) [inline doc/assets.md]: (a good example of a line that is a tad long. It's 120 characters, as far as I know).

**Comment by bdunogier** [comment]
As a side note, this is hard to review on a mobile with those long lines :D

**Comment by DominikaK** [comment]
@bdunogier, I've tried to retain line breaks that would be satisfactory for both sides of the discussion.

As a side note, I would propose a 'rule' somewhat like this in the style guide: "Keep line length to no more than 120 characters. When possible, try to break lines at the end of a sentence or in another logical place". Are you be OK with that?

**Comment by DominikaK** [comment]
@bdunogier: FYI, the lines in style guides mentioned above are public now in https://doc.ezplatform.com/en/1.11/community_resources/contributing/#writing-guidelines-short-version

**Comment by andrerom** [comment]
@lolautruche ?

**Comment by lolautruche** [comment]
+1
